### PR TITLE
refactor session close view to store no shows

### PR DESCRIPTION
**Pull request by timohuber**

### CODE REVIEW

File: pool/app/pool_common/locales/i18n_en.ml [hunk 1]
```diff
@@ -268,7 +268,7 @@ The following follow-up sessions exist:|}
   | SessionCancelMessage ->
     "This reason will be provided to all contacts assigned to this session."
   | SessionClose ->
-    {|S: the contact showed up
+    {|S: the contact did not show up
```
Comment (mabiede) [inline pool/app/pool_common/locales/i18n_en.ml:271]: Also `NS` like above?

```suggestion
    {|NS: the contact did not show up
```

File: pool/cqrs_command/assignment_command.ml [hunk 2]
```diff
@@ -10,6 +10,23 @@ let assignment_effect action id =
     )
 ;;
 
+let update_session_participation_counts contact no_show participated =
+  let open Contact in
+  let open Assignment in
+  let { num_show_ups; num_no_shows; num_participations; _ } = contact in
```
Comment (mabiede) [inline pool/cqrs_command/assignment_command.ml:16]: move to function header

```ocaml
({ Contact.num_show_ups; num_no_shows; num_participations; _ } as contact)
```

**Review by mabiede — CHANGES_REQUESTED** [review]

File: pool/app/pool_common/locales/i18n_en.ml (hunk 1 above)
Comment (timohuber) [reply pool/app/pool_common/locales/i18n_en.ml:271]: Thanks for spotting that

File: pool/cqrs_command/assignment_command.ml [hunk 3]
```diff
@@ -158,24 +184,48 @@ end = struct
...
+                   CCList.filter (fun assignment ->
+                     CCOption.is_none assignment.Assignment.canceled_at)
+                   %> CCList.length
+                   %> NumberOfAssignments.decrement num_assignments)
+            | _, _ -> num_assignments
+          in
+          let contact = { contact with num_assignments } in
+          let mark_as_deleted =
+            match cancel_followups, follow_ups with
+            | true, Some follow_ups ->
```
[the remark this answers is not in the document]
Comment (timohuber) [reply pool/cqrs_command/assignment_command.ml:215]: No, not without worsening readability. I cannot merge them in one boolean as I need to match with `follow_ups`.

File: pool/cqrs_command/assignment_command.ml [hunk 4]
```diff
@@ -262,47 +312,65 @@ end = struct
...
     Logs.info ~src (fun m -> m ~tags "Handle command MarkAsDeleted");
     let* (_ : unit list) =
-      CCList.map Assignment.is_deletable assignments |> CCList.all_ok
+      CCList.map is_deletable assignments |> CCList.all_ok
     in
     let mark_as_deleted =
       CCList.map
-        (fun assignment ->
-          Assignment.MarkedAsDeleted assignment |> Pool_event.assignment)
+        (fun assignment -> MarkedAsDeleted assignment |> Pool_event.assignment)
```
[the remark this answers is not in the document]
Comment (timohuber) [reply pool/cqrs_command/assignment_command.ml:332]: Not it is

File: pool/web/handler/admin_experiments_assignments.ml [hunk 5]
```diff
@@ -136,8 +136,13 @@ let mark_as_deleted req =
       Assignment.find_with_follow_ups database_label assignment_id
     in
     let events =
-      Cqrs_command.Assignment_command.MarkAsDeleted.handle ~tags assignments
-      |> Lwt.return
+      match assignments with
```
[the remark this answers is not in the document]
Comment (timohuber) [reply pool/web/handler/admin_experiments_assignments.ml:139]: No, this map is used to get the contact of the first assignment.

File: pool/database/migration/migration_202303291025.ml [hunk 6]
```diff
@@ -0,0 +1,24 @@
+let add_no_show_count_to_contacts =
+  Sihl.Database.Migration.create_step
+    ~label:"add no show count to contacts"
+    {sql|
+      BEGIN NOT ATOMIC
```
Comment (mabiede) [inline pool/database/migration/migration_202303291025.ml:5]: @timohuber as discussed, split into separate steps.

**Review by mabiede — APPROVED** [review]

**Pull request merged**